lib.sh

```diff
@@ -511,7 +511,14 @@ then
     mkdir -p "$3"
 fi
     rm -f "$3"/"$2"
-    curl -sfL "$1"/"$2" -o "$3"/"$2"
+    if ! curl -sfL "$1"/"$2" -o "$3"/"$2"
+    then
+        msg_box "It seems like the server for the download isn't reachable, or that a temporary error occurred, please try again later."
+        if yesno_box_yes "Do you want to exit this script?"
```

Comment:
We shouldn't allow to not exit here, imho. Because it probably would break some scripts.

Reply:
that's why it's a question, we could add a warning like: "CAUTION, answering 'yes' will break the current operation!"

Reply:
Made it a while loop instead